x-pack/platform/plugins/shared/streams/server/lib/streams/assets/asset_client.ts

```diff
@@ -184,25 +184,28 @@ export class AssetClient {
     }
   ) {}
 
-  async linkAsset(name: string, link: AssetLinkRequest) {
+  async linkAsset(name: string, link: AssetLinkRequest): Promise<AssetLink> {
     const document = toStorage(name, link);
 
     await this.clients.storageClient.index({
       id: document[ASSET_UUID],
       document,
     });
+
+    return toAssetLink(name, link);
   }
 
   async syncAssetList(
     name: string,
-    links: AssetLinkRequest[]
+    links: AssetLinkRequest[],
+    assetType?: AssetType
```

Comment:
current workaround because syncing asset list was overriding the queries (removing all)